python/pyspark/sql/internal.py

```diff
@@ -130,3 +130,42 @@ def make_interval(unit: str, e: Union[Column, int, float]) -> Column:
             "SECOND": "secs",
         }
         return F.make_interval(**{unit_mapping[unit]: F.lit(e)})
+
+    @staticmethod
+    def get_vector(vec: Column, idx: Column) -> Column:
+        unwrapped = F.unwrap_udt(vec)
+        is_dense = unwrapped.getField("type") == F.lit(1)
+        values = unwrapped.getField("values")
+        size = F.when(is_dense, F.array_size(values)).otherwise(unwrapped.getField("size"))
+        sparse_idx = InternalFunction.array_binary_search(unwrapped.getField("indices"), idx)
+        value = (
+            F.when(is_dense, F.get(values, idx))
+            .when(sparse_idx >= 0, F.get(values, sparse_idx))
+            .otherwise(F.lit(0.0))
+        )
+
+        return F.when((0 <= idx) & (idx < size), value).otherwise(
+            F.raise_error(F.printf(F.lit("Vector index must be in [0, %s), but got %s"), size, idx))
+        )
+
+    @staticmethod
+    def array_argmax(arr: Column) -> Column:
```

Comment:
Wouldn't maybe array_position and array_max together do the same thing?
(array_position returns a 1-based index, so you'd subtract 1)